python_modules/dagster/dagster/core/executor/step_delegating/step_delegating_executor.py

```diff
@@ -212,13 +215,37 @@ def execute(self, plan_context: PlanOrchestrationContext, execution_plan: Execut
                         )
 
                 for step in active_execution.get_steps_to_execute():
-                    running_steps[step.key] = step
-                    self._log_new_events(
-                        self._step_handler.launch_step(
-                            self._get_step_handler_context(plan_context, [step], active_execution)
-                        ),
+                    steps_waiting_for_launch[step.key] = step
+
+                launched_steps = []
+                for step in steps_waiting_for_launch.values():
+                    if plan_context.instance.check_can_launch_step(step):
+                        launched_steps.append(step.key)
+                        running_steps[step.key] = step
+                        self._log_new_events(
+                            self._step_handler.launch_step(
+                                self._get_step_handler_context(
+                                    plan_context, [step], active_execution
+                                )
+                            ),
+                            plan_context,
+                            running_steps,
+                        )
+
+                for step_key in launched_steps:
+                    del steps_waiting_for_launch[step_key]
+
+                if steps_waiting_for_launch:
+                    yield DagsterEvent.engine_event(
```

Comment:
this will be way too noisy. Probably we'll actually want to to have some sort of backoff for not allowed steps, or at least have a fixed slower interval for them

Reply:
agree that we likely want a different interval here / not to spam the event log with these (would we ever consider being noisier in the python logs than in the event logs? )